modules/atomic-widgets/validators/styles-validator.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Validators;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Styles_Validator {
+
+	private array $schema;
+
+	public function __construct( array $schema ) {
+		$this->schema = $schema;
+	}
+
+	public static function make( array $schema ): self {
+		return new static( $schema );
+	}
+
+	/**
+	 * @param array $styles
+	 * The key of each item represents the style id,
+	 * and the value is the style object to validate
+	 *
+	 * @return array{
+	 *     0: bool,
+	 *     1: array<string, mixed>,
+	 *     2: array<string>
+	 * }
+	 */
+	public function validate( array $styles ): array {
+		$errors = [];
+
+		foreach ( $styles as $style_id => $style ) {
+			try {
+				$this->validate_id( $style['id'] );
+			} catch ( \Exception $e ) {
+				$errors[] = 'id';
+			}
+
+			foreach ( $style['variants'] as $variant_index => $variant ) {
+				try {
+					$this->validate_meta( $variant['meta'] );
+				} catch ( \Exception $e ) {
+					$errors[] = 'meta';
+				}
+
+				[,$validated_props, $variant_errors] = Props_Validator::make( $this->schema )->validate( $variant['props'] );
+				$styles[ $style_id ]['variants'][ $variant_index ]['props'] = $validated_props;
```

Comment:
It will be much clearer if you will update new variable and not change the reference